core/src/main/scala/com/carto/analyticstoolbox/spark/SpatialFilterPushdownOptimizations.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2022 Azavea
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.carto.analyticstoolbox.spark
+
+import org.apache.spark.sql.SparkSessionExtensions
+
+import com.carto.analyticstoolbox.spark.rules.SpatialFilterPushdownRules
+
+class SpatialFilterPushdownOptimizations extends (SparkSessionExtensions => Unit) {
+  def apply(e: SparkSessionExtensions): Unit =
+    e.injectOptimizerRule(spark => SpatialFilterPushdownRules)
```

Comment:
We don't really use `spark` arg, how about `_ => SpatialFilterPushdownRules`

#### A note for the future:
There is a nice picture of plan execution and injections (source: https://databricks.com/session/how-to-extend-apache-spark-with-customized-optimizations):

<img width="1409" alt="image" src="https://user-images.githubusercontent.com/4929546/163198256-67ccf79e-c2ac-4533-bd8f-aca0f1a56d5d.png">